delphi-checks/src/main/java/au/com/integradev/delphi/checks/FreeAndNilTObjectCheck.java

```diff
@@ -64,7 +64,7 @@ private static boolean isViolation(PrimaryExpressionNode expression) {
       RoutineNameDeclaration routine = (RoutineNameDeclaration) declaration;
       if (routine.fullyQualifiedName().equals("System.SysUtils.FreeAndNil")) {
         ArgumentListNode argumentList = (ArgumentListNode) expression.getChild(1);
-        ExpressionNode argument = argumentList.getArguments().get(0);
+        ExpressionNode argument = argumentList.getArgumentNodes().get(0).getExpression();
```

Comment:
Could be good to have an early exit here if the routine has the wrong number of arguments - although it's unlikely because it's an invocation of `System.SysUtils.FreeAndNil`, it's sort of relying on an outside contract of what that routine should look like.